ambry-rest/src/main/java/com.github.ambry.rest/NettyRequest.java

```diff
@@ -434,6 +458,17 @@ protected void writeContent(AsyncWritableChannel writeChannel, ReadIntoCallbackW
   }
 
   /**
+   * Switches auto reading from the channel on and off.
+   * @param autoRead {@code true} if auto read has to be switched on. {@code false} otherwise.
+   */
+  protected void setAutoRead(boolean autoRead) {
+    channel.config().setAutoRead(autoRead);
+    channel.config().setMaxMessagesPerRead(autoRead ? savedMaxMessagesPerRead : 1);
```

Comment:
Will sync up with you on exactly what setting maxmessages means  - but do you have to set this before setting autoread to true (as in does setting autoread to true potentially trigger things asynchronously?)